data-prepper-core/src/main/java/com/amazon/dataprepper/parser/model/DataPrepperConfiguration.java

```diff
@@ -33,20 +31,7 @@
 
     public static final DataPrepperConfiguration DEFAULT_CONFIG = new DataPrepperConfiguration();
 
-    /**
-     * Construct a DataPrepperConfiguration from a yaml file
-     * @param file
-     * @return
-     */
-    public static DataPrepperConfiguration fromFile(File file) {
-        try {
-            return OBJECT_MAPPER.readValue(file, DataPrepperConfiguration.class);
-        } catch (IOException e) {
-            throw new IllegalArgumentException("Invalid DataPrepper configuration file.", e);
-        }
-    }
-
-    private DataPrepperConfiguration() {}
+    public DataPrepperConfiguration() {}
```

Comment:
Do we need this constructor ? Is this required to unit testing the class ? If not we can make line 32 private and expose method to get default config.

Reply:
The default constructor is used if no configuration is provided.
https://github.com/sbayer55/data-prepper/blob/di-core/data-prepper-core/src/main/java/com/amazon/dataprepper/parser/config/DataPrepperAppConfiguration.java#L55

Before this PR the default configuration was created here: https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-core/src/main/java/com/amazon/dataprepper/parser/model/DataPrepperConfiguration.java#L34